resources/postgres/postgres.go

```diff
@@ -165,15 +165,30 @@ func initDB(pgBinDir, pgDataDir, pgUser string) error {
 	if err != nil && !os.IsNotExist(err) {
 		return fmt.Errorf("failed to check postgres version in data directory path %q, error: %w", versionPath, err)
 	}
-	// Skip initDB if setup already.
+	alreadySetup := false
 	if err == nil {
-		return nil
+		alreadySetup = true
 	}
 
-	if err := os.MkdirAll(pgDataDir, os.ModePerm); err != nil {
+	if err := os.MkdirAll(pgDataDir, 0750); err != nil {
```

Comment:
please add a comment to explain the backstory.

Reply:
Done.